homeassistant/helpers/script.py

```diff
@@ -484,14 +485,24 @@ async def async_run_sequence(iteration, extra_msg="", extra_vars=None):
     async def _async_choose_step(self):
         """Choose a sequence."""
         # pylint: disable=protected-access
-        for conditions, script in await self._script._async_get_choose_data(self._step):
+        choose_data = await self._script._async_get_choose_data(self._step)
+
+        for conditions, script in choose_data["choices"]:
             if all(condition(self._hass, self._variables) for condition in conditions):
                 task = self._hass.async_create_task(
                     script.async_run(self._variables, self._context)
                 )
                 await self._async_run_long_action(task)
                 break
 
+        else:
```

Comment:
Would it be better to change the `break` to a `return`, then you wouldn't need to use `else`?